src/NServiceBus.Core.Tests/Faults/MoveFaultsToErrorQueueTests.cs

```diff
@@ -12,89 +12,140 @@ namespace NServiceBus.Core.Tests
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task ShouldEnrichHeadersWithExceptionDetails(TransportTransactionMode transactionMode)
         {
-            var fakeFaultPipeline = new FakeFaultPipeline();
-            var context = CreateContext("someid", fakeFaultPipeline);
+            var context = CreateContext();
+            var behavior = CreateBehavior(transactionMode);
 
-            var behavior = new MoveFaultsToErrorQueueBehavior(
-                new FakeCriticalError(),
-                "error",
-                "public-receive-address");
+            await behavior.Invoke(context, () => { throw new Exception("exception-message"); });
 
-            await behavior.Invoke(context, () => { throw new Exception("testex"); });
+            if (transactionMode != TransportTransactionMode.None)
+            {
+                await behavior.Invoke(context, () => Task.FromResult(0));
+            }
 
-            Assert.AreEqual("public-receive-address", fakeFaultPipeline.MessageSent.Headers[FaultsHeaderKeys.FailedQ]);
-            //exception details
-            Assert.AreEqual("testex", fakeFaultPipeline.MessageSent.Headers["NServiceBus.ExceptionInfo.Message"]);
+            Assert.AreEqual("public-receive-address", faultPipeline.MessageSent.Headers[FaultsHeaderKeys.FailedQ]);
+            Assert.AreEqual("exception-message", faultPipeline.MessageSent.Headers["NServiceBus.ExceptionInfo.Message"]);
         }
-
-        [Test]
-        public async Task ShouldRegisterFailureInfoWhenMessageIsForwarded()
+        
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task ShouldRegisterFailureInfoWhenMessageIsForwarded(TransportTransactionMode transactionMode)
         {
-            var fakeFaultPipeline = new FakeFaultPipeline();
+            var behavior = CreateBehavior(transactionMode);
+            var context = CreateContext("message-id");
 
-            var behavior = new MoveFaultsToErrorQueueBehavior(
-                new FakeCriticalError(),
-                "error",
-                "public-receive-address");
+            await behavior.Invoke(context, () => { throw new Exception("exception-message"); });
 
+            if (transactionMode != TransportTransactionMode.None)
+            {
+                await behavior.Invoke(context, () => Task.FromResult(0));
+            }
 
-            var context = CreateContext("someid", fakeFaultPipeline);
+            var notification = context.GetNotification<MessageFaulted>();
 
+            Assert.AreEqual("message-id", notification.Message.MessageId);
+            Assert.AreEqual("exception-message", notification.Exception.Message);
+        }
+        
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
```

Comment:
The reason is that in `TransportTransaction.None` message is moved to error queue on the first call. So this test would fail in this case. Maybe what is shows is that we need a two behaviors for handling errors in different transaction modes.